hil/ext/switches/n3000.py

```diff
@@ -195,3 +196,8 @@ def get_port_networks(self, ports):
                 networks.append(('vlan/native', native))
             result[k] = networks
         return result
+
+    def disconnect(self):
+        if _should_save('n3000'):
+            self._save_running_config()
+        self._disconnect()
```

Comment:
Maybe factor this bit out into `_dell_base`? the only bit that's different is the parameter to `_should_save`.

Idea: change the implementation of `_should_save` so that instead of a name, it takes a switch object. Then, it gets the name of the module it needs by doing `obj.__class__.__module__`. Then this `disconnect` can be in some superclass and just pass in `self.`